spec/overcommit/hook_context/pre_commit_spec.rb

```diff
@@ -66,6 +66,30 @@
       end
     end
 
+    context 'when only a submodule change is staged' do
+      around do |example|
+        submodule = repo do
+          `git commit --allow-empty -m "Initial commit"`
+        end
+
+        repo do
+          `git submodule add #{submodule} sub &>/dev/null`
+          `git commit -m "Add submodule"`
+          `echo "Hello World" > sub/submodule-file`
+          `git submodule foreach 'git add submodule-file'`
+          `git submodule foreach 'git commit -m "Another commit"'`
+          `git add sub`
+          example.run
+        end
+      end
+
+      it 'keeps staged submodule change' do
+        subject
+        `git diff`.should be_empty
```

Comment:
But `setup_environment` and `cleanup_environment` run _before_ the commit. Are you suggesting something like this?

```ruby
`git commit -m "Submodule update"`
`git show`.should include('sub')
```